technical-reports/format/types.md

```diff
@@ -12,21 +12,42 @@ If an explicit type is set, but the value does not match the expected syntax the
 
 ## Color
 
-Represents a 24bit RGB or 24+8bit RGBA color in the sRGB color space. The `$type` property MUST be set to the string `color`. The `$value` property MUST be a string containing a [hex triplet/quartet](https://www.w3.org/TR/css-color-4/#typedef-hex-color) including the preceding `#` character. To support other color spaces, such as HSL, translation tools SHOULD convert color tokens to the equivalent value as needed.
+Represents a color in the UI. The `$type` property MUST be set to the string `color`. The `$value` property MUST be an object string containing a string `colorSpace`, a numeric `coordinates` array, optional numeric `alpha`, string `colorProfile`, and string `fallback` to support legacy color spaces as needed.
+
+| Key            |      Type      | Required | Description                                                                                                                              |
+| :------------- | :------------: | :------: | :--------------------------------------------------------------------------------------------------------------------------------------- |
+| `colorSpace`   |    `string`    |    Y     | An string representing the color space.                                                                                                  |
+| `colorProfile` |    `string`    |          | String representing url of color profile.                                                                                                |
+| `coordinates`  | `number array` |    Y     | An array of numeric values representing individual color coordinates.                                                                    |
```

Comment:
I think we were leaning toward “channels” being the term. Also just in the interest of being really specific, “an array” doesn’t specify the length of the list. Calling it a “tuple of 3 values” means that the values must always be 3—no more, no less. Just like any color in CSS, I think we should enforce that the length is always 3, no matter what.

```suggestion
| `channels`  | `[number, number, number]` |    Y     | Tuple of 3 numbers representing the 3 components of the color space. The allowed ranges depend on the `colorSpace` specified. |
```

Reply:
Yes, we were leaning towards that! 

The terms `component` and `channel` were used in different capacities in the W3C CSS Color Level docs, sometimes interchangeably. I agree that we should standardize terminology and clarify what this means.

For example in the [CSS Color Level 4](https://www.w3.org/TR/css-color-4/#color-type) color type definition it reads:

> Colors in CSS are represented as a list of color **components**, also **sometimes called “channels**”, representing axises in the color space. Each channel has a minimum and maximum value, and can take any value between those two. Additionally, every color is accompanied by an **alpha component**, indicating how transparent it is, and thus how much of the backdrop one can see through the color.

The [CSS Color Level 5](https://www.w3.org/TR/css-color-5/#intro) spec builds upon Level 4, yet the terminology is still interchangeable.

In [ColorJS](url), the term "coordinates" could be shorthand for component coordinate or channel coordinate if I assume correctly (need to confirm).

Happy to move back to `channels` for now. 

Also, I suggest we adjust wording to include `none` along with numbers in the allowable channel value types, to account for [4.4“Missing” Color Components and the none Keyword](https://www.w3.org/TR/css-color-4/#missing) support.

Reply:
> Also, I suggest we adjust wording to include `none` along with numbers in the allowable channel value types, to account for [4.4“Missing” Color Components and the none Keyword](https://www.w3.org/TR/css-color-4/#missing) support.

Oh smart! I like the idea of aligning with all that great thinking as much as possible

Reply:
Yes, there have been interest in this support in the legacy color thread!

Reply:
Oh also I think as I said in another comment, maybe we like `components` better and that’s fine with me! I don’t have any strong opinions. We could decide on the exact words at the last minute without it changing the proposal too much.

Reply:
> Calling it a “tuple of 3 values” means that the values must always be 3—no more, no less.

This should be an array of arbitrary length.
CMYK for example has 4 channels.
Other print profiles might have many more for spot colors and varnishes.

Defining these as arrays now means that implementers are less likely to make assumptions that these will always have a length of 3.